### PR TITLE
Feature/reclaim logs

### PR DESCRIPTION
**Pull request by FurkanKayar55**
## Changes

- Added logs for reclaim related actions

### CODE REVIEW

File: LBH.AdultSocialCare.Api/V1/Gateways/CarePackages/Concrete/CarePackageHistoryGateway.cs [hunk 1]
```diff
@@ -38,9 +39,17 @@ public async Task<IEnumerable<CarePackageHistoryDomain>> ListAsync(Guid carePack
...
-            _dbContext.CarePackageHistories.Add(newCarePackageHistory);
+            await _dbContext.CarePackageHistories.AddAsync(newCarePackageHistory);
+            try
+            {
+                await _dbContext.SaveChangesAsync();
+            }
+            catch (Exception ex)
+            {
+                throw new DbSaveFailedException($"Could not save care package history to database {ex.Message}");
+            }
```
Comment (dunnudge) [inline LBH.AdultSocialCare.Api/V1/Gateways/CarePackages/Concrete/CarePackageHistoryGateway.cs:52]: Could you do the save in use case where history is being added and leave save out of gateway? The save changes in use case should save added history as well

File: LBH.AdultSocialCare.Api/V1/UseCase/CarePackages/Concrete/ChangeCarePackageReclaimsStatusUseCase.cs [hunk 2]
```diff
@@ -30,6 +34,15 @@ public async Task<CarePackageReclaimDomain> ExecuteAsync(Guid reclaimId)
...
             await _dbManager.SaveAsync();
+
+            var history = new CarePackageHistory
+            {
+                CarePackageId = reclaim.CarePackageId,
+                Description = $"{reclaim.Type.GetDisplayName()} {ReclaimStatus.Cancelled}",
+            };
+
+            await _carePackageHistoryGateway.Create(history);
+
```
Comment (dunnudge) [inline LBH.AdultSocialCare.Api/V1/UseCase/CarePackages/Concrete/ChangeCarePackageReclaimsStatusUseCase.cs:45]: Add history then invoke save. Removes need to add save in gateway

File: LBH.AdultSocialCare.Api/V1/UseCase/CarePackages/Concrete/EndCarePackageReclaimUseCase.cs [hunk 3]
```diff
@@ -32,6 +35,15 @@ public async Task<CarePackageReclaimDomain> ExecuteAsync(Guid reclaimId, CarePac
...
             await _dbManager.SaveAsync();
+
+            var history = new CarePackageHistory
+            {
+                CarePackageId = reclaim.CarePackageId,
+                Description = $"{reclaim.Type.GetDisplayName()} {ReclaimStatus.Ended}",
+            };
+
+            await _carePackageHistoryGateway.Create(history);
+
```
Comment (dunnudge) [inline LBH.AdultSocialCare.Api/V1/UseCase/CarePackages/Concrete/EndCarePackageReclaimUseCase.cs:46]: History before save

**Review by dunnudge — APPROVED** [review]

File: LBH.AdultSocialCare.Api/V1/Gateways/CarePackages/Concrete/CarePackageHistoryGateway.cs (hunk 1 above)
Comment (FurkanKayar55) [reply LBH.AdultSocialCare.Api/V1/Gateways/CarePackages/Concrete/CarePackageHistoryGateway.cs:52]: make sense, miss that!

**Pull request merged**